brave/brave-core
# feat(wallet): Permissions Duration UI

## Description 
- Updated the `Connect to site` UI to the new V2 design
- Can now only select one account to connect to a `DApp` at a time
- Can now select a `Permissions Duration` for how long you want to allow a `DApp` to have connections with you wallet.


Resolves <https://github.com/brave/brave-browser/issues/28841>



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Go to https://pancakeswap.finance/ , click `Connect Wallet` and select `Brave Wallet`
2. The new `Connect` panel should pop open.
3. You should only be able to `Select` 1 account at a time.
4. Select an `Account` and click `Next`, you should see the `Permission duration` screen.
5. Select `Until I close this site` and click `Connect`
6. In a new tab go to brave://settings/content/ethereum
7. You should see https://pancakeswap.finance/ as an allowed connection.
8. Close the `Tab` for PancakeSwap and the allowed connection should be removed.

https://user-images.githubusercontent.com/40611140/227562651-793e0c78-789a-4d94-9614-d4466e78ebbe.mov
